src/cdomain/value/cdomains/addressDomain.ml

```diff
@@ -110,6 +110,11 @@ struct
     | StrPtr _, UnknownPtr -> None
     | _, _ -> Some false
 
+  let amenable_to_meet x y = match x,y with
+    | StrPtr _, StrPtr _ -> true
+    | Addr x, Addr y when Mval.equal (Mval.top_indices x) (Mval.top_indices y) -> true
```

Comment:
Don't the address set buckets already guarantee this to only be called in such cases because `Mval.top_indices` are the representatives?

Reply:
Even if the buckets guarantee this already, this function can be called with two arbitrary addresses, so we can not just return true.

Reply:
That's true, although it could just be an implementation detail (like the buckets themselves are), which isn't exposed to the outside by the interface. Then such misuse would be impossible.

Reply:
I followed this assumption now to just assume that things placed into the same bucket are amenable to meet, which seems to be the case in our implementation and I think also makes sense semantically.

This largely simplifies the implementation and makes all sorts of extra considerations obsolete.